step-automation-packages/step-automation-packages-manage/src/main/java/step/automation/packages/AutomationPackageServices.java

```diff
@@ -0,0 +1,84 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * STEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package step.automation.packages;
+
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.annotation.PostConstruct;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.UriInfo;
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import step.core.deployment.AbstractStepServices;
+import step.framework.server.security.Secured;
+
+import java.io.InputStream;
+
+@Path("/autopackages")
+@Tag(name = "Automation packages")
+public class AutomationPackageServices extends AbstractStepServices {
+
+    protected AutomationPackageManager automationPackageManager;
+
+    @PostConstruct
+    public void init() throws Exception {
+        super.init();
+        automationPackageManager = getContext().get(AutomationPackageManager.class);
+    }
+
+    @GET
+    @Path("/{name}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Secured(right = "autopack-read")
+    public AutomationPackage getAutomationPackage(@PathParam("name") String automationPackageName) {
+        return automationPackageManager.getAutomationPackageByName(automationPackageName);
+    }
+
+    @DELETE
+    @Path("/{name}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Secured(right = "autopack-write")
+    public void deleteAutomationPackage(@PathParam("name") String automationPackageName) {
+        automationPackageManager.removeAutomationPackage(automationPackageName);
+    }
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Secured(right = "autopack-write")
+    public String createAutomationPackage(@FormDataParam("file") InputStream uploadedInputStream,
+                                          @FormDataParam("file") FormDataContentDisposition fileDetail,
+                                          @Context UriInfo uriInfo) throws Exception {
+        return automationPackageManager.createAutomationPackage(uploadedInputStream, fileDetail.getFileName(), getObjectEnricher());
+
+    }
+
+    @PUT
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("")
+    @Secured(right = "autopack-write")
+    public String updateAutomationPackage(@FormDataParam("file") InputStream uploadedInputStream,
```

Comment:
For the PUT we should add the name to the path.
The current implementation of automationPackageManager.updateAutomationPackage ensures that the package with the provided name already exists. It would be nice to also support the "implicit" creation if it doesn't exist and returns following status depending on if the package had to be created or not:
200 (Service): OK
201 (Service): Created

Reply:
@jeromecomte 
Agree, but in this case how can we resolve two sources of package name (one from path and one from yaml descriptor inside the package)?

Reply:
@iegorov777 I see the point. Not critical for now. We can discuss this quickly tomorrow

Reply:
@jeromecomte 
I changed the logic for PUT, so now it returns the 201 status code if the new package is created. I think, we can merge this implementation to the epic branch.
FYI: also I've merged the master branch to epic, so all changes regarding the 'DataSet' and 'ForEach' operators should already be there.